json/DataTypes/DecodeStudentLong.roc

```diff
@@ -0,0 +1,81 @@
+module [getListStudents]
+
+import JsonData exposing [Json]
+import Student exposing [Student, Module]
+
+JsonErrors : [
+    FieldNotFound Str,
+    ExpectedJsonObject Str,
+    ExpectedJsonArray Str,
+]
+
+getListStudents : Json -> Result (List Student) JsonErrors
+getListStudents = \json ->
+    when json is
+        Object obj ->
+            when Dict.get obj "students" is
+                Ok students -> checkStudentList students
+                _ -> Err (FieldNotFound "Expected field with name students in object")
+
+        _ -> Err (ExpectedJsonObject "Expected an Object")
+
+checkStudentList : Json -> Result (List Student) JsonErrors
+checkStudentList = \json ->
+    when json is
+        Arr students ->
+            List.mapTry (List.map students checkStudentObj) \x -> x
```

Comment:
This can be shortened a bit!

```suggestion
            List.mapTry students checkStudentObj
```

Reply:
Oh nice!  When I was writing it, I felt like I must be something wrong, seeing as I was just passing the identity function